wordpress-mobile/WordPress-iOS
# Notification Comments: add wrapper view controller to display details

Ref: #17790 

To facilitate showing a `NoResultsView` when loading Comments for Notifications, a new `NotificationCommentDetailViewController` has been added. It is used to load/fetch the Comment for display, and show the `CommentDetailViewController` when the Comment has been obtained. 

- The logic for loading/fetching Comments and displaying `CommentDetailViewController` has been moved from `NotificationCommentDetailCoordinator` into the new VC.
- Notification navigation button logic has been moved from `CommentDetailViewController` into the new VC.
- Showing the `NoResultsView` with the notification navigation buttons proved persnickity, so I'll do that in a follow-up.
  - Because there is no loading view, sometimes there is a delay and/or a blank view is displayed after selecting a notification while the Comment is loaded.
- Sometimes when a Comment is fetched, there's a gap at the top of the details view. Once the Comment is cached, and the comment is redisplayed, the layout is fine. Pretty sure I've seen it before, so I think it's legacy. I'll look at it separately. 

To test:
TLDR: everything should work as it did before. 😄 

- Go to Notifications.
- Select a Comment notification.
- Verify the Edit, Previous, Next nav bar buttons work as expected.
- Verify the Previous/Next buttons are disabled when viewing the first/last notification.
- Trash/Spam a comment. Verify the Notification is removed from the list.
- Delete a comment. Verify the Notification is removed from the list.
- Verify My Site Comments look & behave correctly.

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
